python/lsst/summit/utils/bestEffort.py

```diff
@@ -139,20 +139,26 @@ def _parseExpIdOrDataId(expIdOrDataId, **kwargs):
         Parameters
         ----------
         expIdOrDataId : `int` or `dict
-            The exposure id as an int or the dataId as as dict.
+            The exposure id as an int or the dataId as as dict, or an expRecord
+            or a dataCoordinate.
 
         Returns
         -------
         dataId : `dict`
             The sanitized dataId.
         """
-        if type(expIdOrDataId) == int:
+        if isinstance(expIdOrDataId, int):
             _dataId = {'expId': expIdOrDataId}
-        elif type(expIdOrDataId) == dict:
+            _dataId.update(kwargs)
+        elif isinstance(expIdOrDataId, dict):
             _dataId = expIdOrDataId
             _dataId.update(kwargs)
+        elif isinstance(expIdOrDataId, dafButler.DimensionRecord):
+            return expIdOrDataId.dataId
+        elif isinstance(expIdOrDataId, dafButler.DataCoordinate):
+            return expIdOrDataId
         else:
-            raise RuntimeError(f"Invalid expId or dataId type {expIdOrDataId}")
+            raise RuntimeError(f"Invalid expId or dataId type {expIdOrDataId}: {type(expIdOrDataId)}")
```

Comment:
If you're interested, you can also write this as
```
match expIdOrDataId:
    case int() as expId:
        return {"expId": expId, **kwargs}
    case dafButler.DataCoordinate() as dataId:
        assert not kwargs
        return dataId
    case dafButler.DimensionRecord() as record:
        assert not kwargs
        return record.dataId
    case dict() as dataId:
        return {dataId, **kwargs}
raise RuntimeError(...)
```
I would at least recommend adding something like the `assert`s above to guard against `kwargs` being silently ignored unexpected.  Or you could use `DataCoordinate.standardize` to try to include them in the result in all cases.

Reply:
I actually have no idea why that function is trying to stick the kwargs in, when I'm just passing `**kwargs` to all the butler calls directly. I've replaced this with the `match` syntax, but not done the asserts, and instead stopped the function taking kwargs in the first place. Elsewhere I am indeed now using `DataCoordinate.standardize` and as I change more of the util code to use native types, hopefully I can drop nonsense like this entirely, in time 🙂